### PR TITLE
CPDRP-1123 show provider withdrawals to sits

### PR DESCRIPTION
**Pull request by xswoodsx**
## Ticket and context

Ticket:

## Tech review
Ticket [1123](https://dfedigital.atlassian.net/jira/software/projects/CPDRP/boards/102?assignee=5f3543bb3e9e2e004db3189d&selectedIssue=CPDRP-1123)

FYI - There's no longer a requirement in this ticket to give a SIT the functionality to delete a withdrawn user. 

### Code quality checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.
 
Use cpd-test+tutor-10@example.com to see the user that has been withdrawn

### CODE REVIEW

**Comment by xswoodsx** [comment]
I'm not confident in my set participant categories set up, i.e. the where.not on each training status. It seemed the quickest way to remove these users. Also, in when to show the withdrawn ppts, i.e cip eligible ppts array.

File: spec/services/set_participant_categories_spec.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@
     let!(:details_being_checked_ect) { create(:participant_profile, :ect, :ecf_participant_eligibility, :ecf_participant_validation_data, school_cohort: school_cohort) }
     let!(:primary_mentor) { create(:participant_profile, :mentor, :ecf_participant_eligibility, :ecf_participant_validation_data, :primary_profile, school_cohort: school_cohort) }
     let!(:secondary_mentor) { create(:participant_profile, :mentor, :ecf_participant_eligibility, :ecf_participant_validation_data, :secondary_profile, school_cohort: school_cohort) }
+    let!(:withdrawn_ect) { create(:participant_profile, :ect, :ecf_participant_eligibility, :ecf_participant_validation_data, training_status: "withdrawn", school_cohort: school_cohort) }
```
Comment (ngw) [inline spec/services/set_participant_categories_spec.rb:16]: You can create lists with FactoryBot, I see you populate 2 variables maybe it's not necessary? https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#building-or-creating-multiple-records

File: app/services/set_participant_categories.rb [hunk 2]
```diff
@@ -27,41 +27,49 @@ def set_participant_categories
...
+    ParticipantCategories.new(cip_eligible_participants, [], withdrawn_participants, contacted_for_info_participants, [])
   end
 
   def active_participant_profiles
     ParticipantProfilePolicy::Scope.new(user, school_cohort.active_ecf_participant_profiles).resolve
   end
 
   def ineligible_participants
-    active_participant_profiles.ineligible_status.includes(:user).order("users.full_name")
+    active_participant_profiles.ineligible_status.includes(:user).order("users.full_name").where.not(training_status: "withdrawn")
```
Comment (ngw) [inline app/services/set_participant_categories.rb:46]: I don't see anything particularly wrong in your use of `not`, but probably I'd personally use a scope mostly to have a clearer name ('acknowledged'? 'admitted'?), but it's a nitpick + personal taste, nothing else.

Comment (ngw) [reply app/services/set_participant_categories.rb:46]: I'm more puzzled by your use of "withdrawn" as a string instead of using a symbol.

**Review by ngw — APPROVED** [review]

File: app/services/set_participant_categories.rb (hunk 2 above)
Comment (xswoodsx) [reply app/services/set_participant_categories.rb]: Yeah, i should have used a symbol. I'll change it.

File: app/services/set_participant_categories.rb [hunk 3]
```diff
@@ -27,41 +27,49 @@ def set_participant_categories
...
+    active_participant_profiles.contacted_for_info.includes(:user).order("users.full_name").where.not(training_status: :withdrawn)
   end
 
   def details_being_checked_participants
-    active_participant_profiles.details_being_checked.includes(:user).order("users.full_name")
+    active_participant_profiles.details_being_checked.includes(:user).order("users.full_name").where.not(training_status: :withdrawn)
   end
 
   def fip_flag_active_ineligible_participants
     ineligible_participants - eligible_participants
```
Comment (ethanmills) [inline app/services/set_participant_categories.rb:66]: Where does a withdrawn ERO mentor appear?

Comment (xswoodsx) [reply app/services/set_participant_categories.rb:66]: They should appear in the withdrawn section. I've changed the training status on "Previous Participation ERO" user test data and they show up there.

**Review by ethanmills — APPROVED** [review]

**Pull request merged**